hooks/tk-multi-publish2/basic/start_version_control.py

```diff
@@ -0,0 +1,355 @@
+        """
+
+        publisher = self.parent
+        version_number = None
+
+        work_file_template = item.properties.get("work_file_template")
+        if work_file_template:
+            if work_file_template.validate(path):
+                self.logger.debug(
+                    "Using work file template to determine version number.")
+                work_file_fields = work_file_template.get_fields(path)
+                if "version" in work_file_fields:
+                    version_number = work_file_fields.get("version")
+            else:
+                self.logger.debug(
+                    "Work file template did not match path")
+        else:
+            self.logger.debug(
+                "Work file template unavailable for version extraction.")
+
+        if version_number is None:
+            self.logger.debug(
+                "Using path info hook to determine version number.")
+            version_number = publisher.util.get_version_number(path)
+
+        return version_number
+
+def _session_path():
+    """
+    Return the path to the current session
+    :return:
+    """
+    path = cmds.file(query=True, sn=True)
+
+    if isinstance(path, unicode):
+        path = path.encode("utf-8")
+
+    return path
+
+
+def _save_session(path):
+    """
+    Save the current session to the supplied path.
+    """
+
+    # Maya can choose the wrong file type so we should set it here
+    # explicitly based on the extension
+    maya_file_type = None
+    if path.lower().endswith(".ma"):
+        maya_file_type = "mayaAscii"
+    elif path.lower().endswith(".mb"):
+        maya_file_type = "mayaBinary"
+
+    cmds.file(rename=path)
+
+    # save the scene:
+    if maya_file_type:
+        cmds.file(save=True, force=True, type=maya_file_type)
+    else:
+        cmds.file(save=True, force=True)
+
+
+# TODO: method duplicated in all the maya hooks
+def _get_save_as_action():
+    """
+
+    Simple helper for returning a log action dict for saving the session
+    """
+
+    engine = sgtk.platform.current_engine()
+
+    # default save callback
+    callback = cmds.SaveScene
+
+    # if workfiles2 is configured, use that for file save
+    if "tk-multi-workfiles2" in engine.apps:
+        app = engine.apps["tk-multi-workfiles2"]
+        if hasattr(app, "show_file_save_dlg"):
+            callback = app.show_file_save_dlg
+
+    return {
+        "action_button": {
+            "label": "Save As...",
+            "tooltip": "Save the current session",
+            "callback": callback
+        }
+    }
+
+
+def _get_version_docs_action():
+    """
+    Simple helper for returning a log action to show version docs
+    """
+    return {
+        "action_open_url": {
+            "label": "Version Docs",
+            "tooltip": "Show docs for version formats",
+            "url": "https://support.shotgunsoftware.com/hc/en-us/articles/115000068574-User-Guide-WIP-#What%20happens%20when%20you%20publish"
+        }
+    }
```

Comment:
comment regarding the maya icon below (couldn't comment on it!) - do we need it? could we use the engine icon instead?

Reply:
also now that we are adding publish icons to the engine, wondering if we either need subfolders e.g. `resources/tk-multi-publish2` or perhaps `hooks/tk-multi-publish2/icons` or something? I think i would prefer the latter out of the two...

Reply:
Agree. This makes the plugins more portable as well since they don't have to know about the engine structure. I've made the change and included all the icons the hooks need in `hooks/tk-multi-publish2/icons/`